src/main/java/me/jellysquid/mods/sodium/client/render/occlusion/BlockOcclusionCache.java

```diff
@@ -35,7 +35,7 @@ public boolean shouldDrawSide(BlockState selfState, BlockView view, BlockPos pos
 
         BlockState adjState = view.getBlockState(adjPos);
 
-        if (selfState.isSideInvisible(adjState, facing)) {
+        if (selfState.isSideInvisible(adjState, facing) || (adjState.hidesNeighborFace(view, pos, adjState, facing.getOpposite()) && selfState.supportsExternalFaceHiding())) {
```

Comment:
I think this change is also needed according to the comment above?

```suggestion
        if (selfState.isSideInvisible(adjState, facing) || (adjState.hidesNeighborFace(view, pos, selfState, facing.getOpposite()) && selfState.supportsExternalFaceHiding())) {
```